plugins/inputs/slurm/slurm.go

```diff
@@ -0,0 +1,478 @@
+//go:generate ../../../tools/config_includer/generator
+//go:generate ../../../tools/readme_config_includer/generator
+package slurm
+
+import (
+	"context"
+	_ "embed"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/common/tls"
+	"github.com/influxdata/telegraf/plugins/inputs"
+
+	goslurm "github.com/pcolladosoto/goslurm/v0038"
+)
+
+//go:embed sample.conf
+var sampleConfig string
+
+type Slurm struct {
+	URL              string          `toml:"url"`
+	Username         string          `toml:"username"`
+	Token            string          `toml:"token"`
+	EnabledEndpoints []string        `toml:"enabled_endpoints"`
+	ResponseTimeout  config.Duration `toml:"response_timeout"`
+	Log              telegraf.Logger `toml:"-"`
+	tls.ClientConfig
+
+	client      *goslurm.APIClient
+	baseURL     *url.URL
+	endpointMap map[string]bool
+}
+
+func (*Slurm) SampleConfig() string {
+	return sampleConfig
+}
+
+func (s *Slurm) Init() error {
+	if len(s.EnabledEndpoints) == 0 {
+		s.EnabledEndpoints = []string{"diag", "jobs", "nodes", "partitions", "reservations"}
+	}
+
+	s.endpointMap = make(map[string]bool{}, len(s.EnabledEndpoints))
+	for _, endpoint := range s.EnabledEndpoints {
+		switch e := strings.ToLower(endpoint) {
+			case "diag", "jobs", "nodes", "partitions", "reservations":
+				s.endpointMap[e] = true
+			default:
+				return fmt.Errorf("unknown endpoint %q, endpoint)
```

Comment:
```suggestion
				return fmt.Errorf("unknown endpoint %q", endpoint)
```

Reply:
Thanks a ton for catching that! I had 'fixed' that on ed20b4dff2d1df22ee4595039cdf7ba91d9a3b40 but was waiting to batch all the commits so as to run the pipelines once... Sorry about that!